### ISSUE TITLE
Strategies will not validate

### ISSUE THREAD

**Issue by deewilcox** [uzyn/cakephp-opauth#12]
I have the Facebook strategy setup, including app ID and app secret. Auth returned the following error in my UsersController.php with this method: 

`public function opauth_complete() {
       debug($this->data);
       print_r($this->data['validated']);
       exit;
    }`

Error message:

`/app/Controller/UsersController.php (line 11)
array(
    'error' => array(
        'provider' => null,
        'code' => 'invalid_auth_missing_components',
        'message' => 'Invalid auth response: Missing key auth response components.'
    ),
    'validated' => false
)`

I also have the Google strategy setup. I receive the following error when the I run Path_to_app/auth/google: 

`````` array(
    'error' => array(
        'provider' => null,
        'code' => 'invalid_auth_missing_components',
        'message' => 'Invalid auth response: Missing key auth response components.'
    ),
    'validated' => false
)```
``````

**Comment by uzyn** [comment]
I can't seem to be able to replicate this though there has been [a few reports](https://groups.google.com/forum/?fromgroups=#!topic/opauth/3IWw3Hu_h-0) of the same issue.

Would you be able to quickly check if there is SSL support at your PHP setup and that `allow_url_fopen` is set to On at php.ini?

**Comment by deewilcox** [comment]
Thanks for your help!

SSL is not enabled for this site, so I'm not sure whether or not it is supported. However, I tried setting allow_url_fopen = On in php.ini and another version for .htacess, and both resulted in server errors. 

I did notice that the $response['error'] array was returning null, so I debugged that. $response itself is returning null, which is causing the error. Any ideas why $response would be empty?

**Comment by uzyn** [comment]
I can't seem to replicate this though I believe you would require OpenSSL to be setup in order to call Facebook API on https://graph.facebook.com

**Comment by deewilcox** [comment]
Thanks for the follow up. I have confirmed that we have OpenSSL enabled. It looks like the param is being returned empty even though the Strategies are set up. Have you heard of any issues with that for other users?

**Comment by uzyn** [comment]
There has been reports on that, but it worked fine for most.

Would you be kind to share your `phpinfo()` output?

Also, did you try any other strategies, eg. Twitter?

**Comment by deewilcox** [comment]
Hi, there. 

I was just about to send a link to you to check the phpinfo() output, but I had to change my Auth settings to allow all in order for that to work. Upon doing so, OPAuth validates. So, the hang-up is making sure Auth allows the appropriate controllers/actions. I had already allowed opauth_complete, auth, and callback. I will check the plugin for additional methods that need to be allowed and report back.

Thanks!
